generic_config_updater/patch_sorter.py

```diff
@@ -573,6 +573,35 @@ def _find_ref_paths(self, paths, config):
             refs.extend(self.path_addressing.find_ref_paths(path, config))
         return refs
 
+class NoEmptyTableMoveValidator:
+    """
+    A class to validate that a move will not result in an empty table, because empty table do not show up in ConfigDB.
+    """
+    def __init__(self, path_addressing):
+        self.path_addressing = path_addressing
+
+    def validate(self, move, diff):
+        simulated_config = move.apply(diff.current_config)
+        op_path = move.path
+
+        if op_path == "": # If updating whole file
+            tables_to_check = simulated_config.keys()
+        else:
+            tokens = self.path_addressing.get_path_tokens(op_path)
+            tables_to_check = [tokens[0]]
+
+        return self._validate_tables(tables_to_check, simulated_config)
+
+    def _validate_tables(self, tables, config):
+        for table in tables:
+            if not(self._validate_table(table, config)):
+                return False
+        return True
+
+    def _validate_table(self, table, config):
+        is_empty = table in config and not(config[table])
+        return not(is_empty)
```

Comment:
>not [](http://example.com/codeflow?start=15&length=3)

too many `not`. How about `table not in config or config[table]`

Reply:
It will look like this:
```python
    def _validate_table(self, table, config):
        return table not in config or config[table]
```

This means the table is either in not in config, and table has content. Why does this mean the table is not empty? It will take future developer a while to understand it. I think it is easier to have a flag `is_empty` and revert the flag on return.

Reply:
Either way is okay to me.
Could you add function level comment "the only invalid case is if table exist and is empty"?

Reply:
[offline discussion] will update and add a small comment

Reply:
updated